src/Controller/CRUDController.php

```diff
@@ -66,11 +68,29 @@ class CRUDController extends Controller
      */
     private $templateRegistry;
 
-    public function setContainer(?ContainerInterface $container = null): void
+    public function setContainer(ContainerInterface $container): ?ContainerInterface
     {
-        $this->container = $container;
+        $result = parent::setContainer($container);
 
         $this->configure();
+
+        return $result;
+    }
+
+    public static function getSubscribedServices()
+    {
+        return array_merge(parent::getSubscribedServices(), [
+            'sonata.admin.pool' => Pool::class,
+            'sonata.admin.breadcrumbs_builder' => BreadcrumbsBuilderInterface::class,
+            'sonata.admin.audit.manager' => AuditManagerInterface::class,
+            'sonata.admin.object.manipulator.acl.admin' => AdminObjectAclManipulator::class,
+            'sonata.exporter.exporter' => '?'.Exporter::class,
+            'sonata.admin.admin_exporter' => '?'.AdminExporter::class,
+            'sonata.admin.security.acl_user_manager' => '?object',
+
+            'logger' => '?'.LoggerInterface::class,
+            'translator' => TranslatorInterface::class,
```

Comment:
This one does not want to work either, and it is failing here: `AddDependencyCallsCompilerPassTest`

Reply:
I registered it and added `TranslatorInterface` as alias: https://github.com/sonata-project/SonataAdminBundle/pull/6476/commits/151470e60451ce01f2b070094d8114d01df2ccdc#diff-c5ec9ed7cdde122e38ecf3c1d2d82263R757-R760